app/src/main/java/com/example/drawingactivity/DialogUtils.kt

```diff
@@ -137,6 +135,223 @@ object DialogUtils {
             }
         }
     }
+
+    fun colorPickerBottomSheet(
+        context: Context,
+        previousColor: FloatArray,
+        /**
+         * Lambda that provides the string of the current color.
+         * Returns nothing. Used to save the hex value of the current color.
+         */
+        onSubmit: ((FloatArray?, String?) -> Unit)
+    ) {
+        val sheet = BottomSheetDialog(context)
+        val view = LayoutInflater.from(context).inflate(R.layout.color_dialog, null)
+        val huePicker = view.findViewById<HueView>(R.id.huePicker)
+        val gradientPicker = view.findViewById<GradientView>(R.id.gradientPicker)
+        val hexText = view.findViewById<ForcableEditText>(R.id.hexText)
+        val hueText = view.findViewById<ForcableEditText>(R.id.hueText)
+        val satText = view.findViewById<ForcableEditText>(R.id.satText)
+        val valText = view.findViewById<ForcableEditText>(R.id.valText)
+        val redText = view.findViewById<ForcableEditText>(R.id.redText)
+        val greenText = view.findViewById<ForcableEditText>(R.id.greenText)
+        val blueText = view.findViewById<ForcableEditText>(R.id.blueText)
+        val topColorSquare = view.findViewById<View>(R.id.topColor)
+        val bottomColorSquare = view.findViewById<View>(R.id.bottomColor)
+        val confirmButton = view.findViewById<Button>(R.id.okButton)
+        confirmButton.setOnClickListener {
+            val submitArray = FloatArray(3)
+            submitArray[0] = (hueText.text.toString().toFloat())
```

Comment:
You can make functions off your edit text so you don't need to duplicate these common operations over and over how I asked Hamdan to do

Reply:
implemented this to turn my 6000 line dialog into something readable by taking this advice